someip2val/src/lib/broker_feeder/actuator_subscriber.cc

```diff
@@ -0,0 +1,173 @@
+            LOG_INFO << "Not connected" << std::endl;
+            if (backoff < 10) {
+                backoff++;
+            }
+            continue;
+        } else {
+            backoff = 1;
+        }
+
+        LOG_INFO << "Connected to [" << client_->GetBrokerAddr() << "]" << std::endl;
+
+        ::kuksa::val::v1::SubscribeRequest request;
+        for (auto i=0; i<actuators_.size(); i++) {
+            auto entry = request.add_entries();
+            entry->set_path(actuators_[i]);
+            entry->add_fields(::kuksa::val::v1::Field::FIELD_ACTUATOR_TARGET);
+            entry->add_fields(::kuksa::val::v1::Field::FIELD_METADATA);
+        }
+
+        ::kuksa::val::v1::SubscribeResponse response;
+        subscriber_context_ = client_->createClientContext();
+        if (LOGGER_ENABLED(LEVEL_DBG)) {
+            { LOG_DEBUG << "Subscribing: [ " << actuators_ << "]" << std::endl; }
+        }
+        std::unique_ptr<::grpc::ClientReader<::kuksa::val::v1::SubscribeResponse>> reader(
+            client_->Subscribe(subscriber_context_.get(), request));
+
+        LOG_INFO << "Actuator targets Subscribed!" << std::endl;
+        while (subscriber_active_ && reader->Read(&response)) {
+            LOG_TRACE << "[SUB] updates_size:" << response.updates_size() << std::endl;
+            // std::pair<std::string, int> entries;
+            ActuatorValues changes;
+            for (auto& update : response.updates()) {
+                if (!subscriber_active_) break;
+                if (LOGGER_ENABLED(LEVEL_DBG)) {
+                    std::stringstream ss;
+                    ss  << "[SUB] " << update.entry().path()
+                        << ", value_case:" << static_cast<int>(update.entry().actuator_target().value_case())
+                        << ", target: { " << update.entry().actuator_target().ShortDebugString() << " }";
+                    LOG_DEBUG << ss.str() << std::endl;
+                }
+                // DatapointEntry dpe;
+                // dpe.name = update.entry().path();
+                // dpe.value = update.entry().actuator_target();
+                changes.emplace(
+                    update.entry().path(),
+                    update.entry().actuator_target());
+            }
+            if (cb_ && subscriber_active_) {
+                cb_(changes);
+            }
+        }
+        grpc::Status status = reader->Finish();
+        if (status.ok()) {
+            LOG_INFO << "Disconnected." << std::endl;
+        } else {
+            LOG_ERROR << "Disconnected with status: "
+                    << subscriber_context_->debug_error_string() << std::endl;
+            if (subscriber_active_) {
+                client_->handleGrpcError(status, "ActuatorSubscriber::Run()");
+                // prevent busy polling if subscribe failed with error
+                std::this_thread::sleep_for(std::chrono::seconds(5));
+            }
+        }
+        {
+            // TODO: lock mutex
```

Comment:
When do we need to fix the TODO. Consequences if we do not fix it?

Reply:
it was actually done :) I'll remove the comment